### PR TITLE
Sync template secrets

### PR DESCRIPTION
**Pull request by stlaz**
Add syncing of template's secrets from the top level config

I accidentally built it on top https://github.com/openshift/cluster-authentication-operator/pull/56 when I did not check the git branch I'm on, I can split it if required.

### CODE REVIEW

File: pkg/operator2/configsync.go [hunk 1]
```diff
@@ -121,48 +126,67 @@ func newSourceDataIDPConfigMap(index int, configMap configv1.ConfigMapNameRefere
 	return dest, ret
 }
 
-func newIDPSyncData() idpSyncData {
-	configMaps := map[string]sourceData{}
-	secrets := map[string]sourceData{}
+func newSourceDataTemplateSecret(secretRef configv1.SecretNameReference, key string) (string, sourceData) {
+	dest := getTemplateName(secretRef.Name, key)
+
+	vol, mount, path := secretVolume(getTemplatePath("template", dest), dest, key)
```
Comment (ericavonb) [inline pkg/operator2/configsync.go]: nit: bit awkward, maybe put a `path := getTem...` below the `dest` var declaration

Comment (stlaz) [reply pkg/operator2/configsync.go]: fixed

**Comment by stlaz** [comment]
Rebased on current master, fixed teh nit.

**Pull request merged**